flake.nix

```diff
@@ -0,0 +1,123 @@
+{
+  description =
+    "tinygrad: For something between PyTorch and karpathy/micrograd. Maintained by tiny corp.";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils }@I:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        P = import nixpkgs {
+          inherit system;
+          config.allowUnfree = true;
+        };
+        B = builtins;
+        L = P.lib;
+
+        pythonVersion = "310";
+        pythonPackageStr = "python${pythonVersion}";
+        pythonNixpkgs = P."${pythonPackageStr}Packages";
+
+        tinygrad = pythonNixpkgs.buildPythonPackage {
+          pname = "tinygrad";
+          version = "v.0.6.0";
+          src = ./.;
+
+          propagatedBuildInputs = with pythonNixpkgs; [
+            networkx
+            numpy
+            pillow
+            pyopencl
+            pyyaml
+            requests
+            tqdm
+          ];
+
+          # TODO(b7r6): enable when all tests pass...
+          doCheck = false;
+          pythonImportsCheck = [ "tinygrad" ];
+        };
+
+        commonBuildInputs = with P; [ git nixfmt ];
+        commonPythonPackages = pypkgs:
+          with pypkgs; [
+            flake8
+            mypy
+            onnx
+            onnxruntime
+            opencv4
+            pylint
+            pytest
+            pytest-xdist
+            python-lsp-server
+            pytorch-bin
+            pyyaml
+            safetensors
```

Comment:
We shouldn't need safetensors here cuz tinygrad supports loading safetensors natively.

Reply:
I found that some of the more important examples don't work under any runtime without some imports like e.g. `safetensors` and `sentencepiece`. There's no objectively right answer here, but I'm biased towards people can `nix develop` and immediately run the marquee models that `tinygrad` supports, e.g. `LLaMA`.

Again, I will defer to your judgement here.

Reply:
Oh yea I see there is a test that depends on safetensors, ok yea safetensors is fine.